src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/custom.py

```diff
@@ -1359,6 +1376,70 @@ def tag(self, tags):
                                                    self.api_version,
                                                    parameters)
 
+    def invoke_action(self, action=None, parameters=None, url=None):
+        """
+        Formats Url if none provided and sends the POST request with the url and parameters.
+        """
+        from msrestazure.azure_operation import AzureOperationPoller
```

Comment:
//CC: @lmazuel, we are accessing private fields like following. Feel free to comment if we need adjustments to get things maintainable. We are having e2e test coverage, so should be fine